packages/synapse-react-client/src/components/ChallengeSubmission/SubmissionDirectoryList.tsx

```diff
@@ -92,16 +92,22 @@ function SubmissionDirectoryList({
   const { data: headerResults, refetch } = useGetEntityChildren(request, {
     enabled: !!challengeProjectId,
     useErrorBoundary: true,
-    onSuccess: data => {
+  })
+
+  useEffect(() => {
+    if (headerResults) {
       const newHeaders = [...fetchedHeaders]
       const headerPage = Math.floor(((page + 1) * PER_PAGE) / HEADERS_PER_PAGE)
       const start = headerPage * HEADERS_PER_PAGE
-      newHeaders.splice(start, start + HEADERS_PER_PAGE, ...data.page)
+      newHeaders.splice(start, start + HEADERS_PER_PAGE, ...headerResults.page)
       setFetchedHeaders(newHeaders)
       setFetchNextPage(false)
-      setNextPageToken(data.nextPageToken)
-    },
-  })
+      setNextPageToken(headerResults.nextPageToken)
+    }
+    // TODO: Temporary useEffect hook to remove onSuccess QueryOption for @tanstack/react-query v5
+    // Refactor this component to remove this effect.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [headerResults])
```

Comment:
Update the state variables when we get a new result.

We could do much better (could use `useInfiniteQuery`, could derive state instead of using state variables), but this is the most simple fix to let us upgrade.